core/templates/dev/head/app.js

```diff
@@ -34,11 +34,18 @@ var oppia = angular.module(
 // Set default headers for POST and PUT requests.
 // Add an interceptor to convert requests to strings and to log and show
 // warnings for error responses.
-oppia.config(['$interpolateProvider', '$httpProvider',
-    function($interpolateProvider, $httpProvider) {
+oppia.config(['$interpolateProvider', '$httpProvider', '$locationProvider',
+  function($interpolateProvider, $httpProvider, $locationProvider) {
   $interpolateProvider.startSymbol('<[');
   $interpolateProvider.endSymbol(']>');
 
+  $locationProvider.html5Mode(false);
+  // The following prevents the page from reloading if the search query is
```

Comment:
Sorry -- one nit. Can you add the word 'search' before 'page'? I think it's clearer that way.